python/lsst/ts/cbp/mock_server.py

```diff
@@ -160,13 +160,20 @@ async def cmd_loop(self, reader, writer):
                         self.log.debug(f"parameter={parameter}")
                     except IndexError:
                         parameter = None
-                    if parameter is None:
-                        msg = await command_method()
-                    else:
-                        try:
+                    try:
+                        if parameter is None:
+                            msg = await command_method()
+                        else:
                             msg = await command_method(parameter)
-                        except Exception:
-                            raise asyncio.TimeoutError()
+                    except ValueError as e:
+                        self.log.info(f"command {line} failed: {e}")
+                        # TODO DM-27693: reply with an error signal so the
+                        # client knows there is a problem
+                    except Exception:
+                        # An unexpected error; log a traceback
+                        self.log.exception("Bug! Command {line} filed.")
+                        # TODO DM-27693: reply with an error signal so the
+                        # client knows there is a problem
```

Comment:
Almost right. Unfortunately you then write `msg` which is the value from the previous iteration of the loop.

There are many ways around that. Here are two I like:
* Put the write/debug/drain in an `else` clause. That only runs if no except is raised.
* Initialize msg to None and only write it if it is not None.

(One I beg you NOT to use is to put a `continue` in each `except` clause, to skip the write/debug/drain. That's far too risky, because forgetting the `continue` will cause very subtle bugs (especially if the first command fails -- which will lead to using an undefined variable!)).